### DISCUSSION TITLE
How get registration flow for browser correctly?

### DISCUSSION THREAD

**Discussion by winston0410** [ory/kratos#1998] [Q&A] [answered]
What is the correct route for initializing Registration Flow for browser? I have tried http://localhost:4433/self-service/registration/browser, but this return 404 in my own UI application instead of the following response 

https://www.ory.sh/kratos/docs/reference/api#operation/initializeSelfServiceRegistrationFlowWithoutBrowser

This is my `kratos.yml`

```
---
version: v0.8.0-alpha.3

dsn: postgres://kratos:secret@postgresd:5432/kratos?sslmode=disable&max_conns=20&max_idle_conns=4

identity:
  default_schema_url: file:///etc/kratos/identity.schema.json
  schemas:
    - id: customer
      url: file:///etc/kratos/identity.schema.json

serve:
  public:
    base_url: http://0.0.0.0:4433/
    cors:
      enabled: true
  admin:
    base_url: http://0.0.0.0:4434/

selfservice:
  default_browser_return_url: http://0.0.0.0:3000/
  flows:
    error:
      ui_url: http://0.0.0.0:3000/error
    registration:
      ui_url: http://0.0.0.0:3000/registration
    login:
      ui_url: http://0.0.0.0:3000/login

  methods:
    password:
      enabled: true
    oidc:
      enabled: true

courier:
  smtp:
    connection_uri: smtps://test:test@mailslurper:1025/?skip_ssl_verify=true
```

Should I remove the `ui_url`?

**Comment by aeneasr** [comment, marked as answer]
The 0.0.0.0 looks incorrect. Where did you find these values? We should correct them if they are in the docs :)

**Reply by winston0410** [reply]
Thanks for the reply! I have used that because of using Docker(which 127.0.0.1 doesn't work and only 0.0.0.0), but I cannot recall why

Anyway I understand how to do it properly now, for someone who has the same issue as i do:

To use a flow(using registration flow, server-side rendering as an example here), you have to:

- Set the ui_url, and then send a request to [init the flow](https://www.ory.sh/kratos/docs/reference/api#operation/initializeSelfServiceRegistrationFlowForBrowsers)

- kratos will 302 the user in the browser to the `ui_url` with `flows` query parameter, with that id you will then [get the flow here](https://www.ory.sh/kratos/docs/reference/api#operation/getSelfServiceRegistrationFlow)

- use the return data to build your UI and submit the request.

**Reply by vinckr** [reply]
I will mark this as answered @winston0410 ,
Thanks for supplying your fix!
